### PR TITLE
Implement APIv1 behaviors

### PR DESCRIPTION
**Pull request by danielsdeleo**

### CODE REVIEW

**Comment by danielsdeleo** [comment]
Have this PR up to fix test issue with ChefFS mode: https://github.com/chef/chef-server/pull/753

File: lib/chef_zero/endpoints/actor_endpoint.rb [hunk 1]
```diff
@@ -8,6 +8,16 @@ module Endpoints
     # /organizations/ORG/users/NAME
     # /users/NAME
     class ActorEndpoint < RestObjectEndpoint
+
+      def get(request)
+        result = super
+        user_data = FFI_Yajl::Parser.parse(result[2], :create_additions => false)
```
Comment (jrunning) [inline lib/chef_zero/endpoints/actor_endpoint.rb]: FWIW RestBase [defines `parse_json` and `to_json` helpers](https://github.com/chef/chef-zero/blob/75e1d346dedbc40cd325c1f0dee2275bcace9b6a/lib/chef_zero/rest_base.rb#L286-L292). (I discovered this too late for my last PR).

Comment (danielsdeleo) [inline lib/chef_zero/endpoints/actor_endpoint.rb]: Nice catch.

**Comment by jrunning** [comment]
:+1:

**Comment by danielsdeleo** [comment]
@chef/client-dev-tools Ready for review.

File: lib/chef_zero/endpoints/principal_endpoint.rb [hunk 2]
```diff
@@ -30,13 +30,22 @@ def get(request)
...
             'authz_id' => '0'*32,
             'org_member' => org_member
-          })
+          }
+
+          response_data =
+            if request.api_v0?
+              principal_data
+            else
+              { "principals" => [ principal_data ] }
```
Comment (jkeiser) [inline lib/chef_zero/endpoints/principal_endpoint.rb]: If there is both a user and a client with the given name, should we return them both?

**Comment by jkeiser** [comment]
@danielsdeleo one question, but looks good!

File: lib/chef_zero/endpoints/principal_endpoint.rb (hunk 2 above)
Comment (danielsdeleo) [inline lib/chef_zero/endpoints/principal_endpoint.rb]: Yeah, that's correct, according to my reading of the code here: https://github.com/chef/chef-server/blob/218579549024b26b3b97c9958503ff3651e3fb16/src/oc_erchef/apps/oc_chef_wm/src/chef_wm_named_principal.erl#L92-L102

OTOH, there's no pedant test for that, and it's an existing bug, and I probably won't have time to fix all that. So my inclination is to leave it as is until the problem comes up again.

**Comment by jkeiser** [comment]
:+1:

**Pull request merged**